src/librustc_codegen_ssa/lib.rs

```diff
@@ -117,6 +116,19 @@ bitflags::bitflags! {
     }
 }
 
+#[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct CrateNum(pub u32);
```

Comment:
This name can be confused with `def_id::CrateNum`.

Reply:
Good point. I chose to keep the name `CrateNum` mainly because I feel `LinkingCrateNum` is misleading. `LinkingCrateNum` gives an impression that this "crate num" is solely for the purpose of linking, but in fact it is just for `CrateInfo` to index multiple crates. In addition, this new `rustc_codegen_ssa::CrateNum` serves the same purpose as `def_id::CrateNum` does under different scope; so I decided to keep its name `CrateNum`. While reviewing this PR, we can think about it more and find it a better name.